src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.Tests.cs

```diff
@@ -350,5 +351,20 @@ private int GetChecksumForFormatSpecificFields(TarEntry entry, TarEntryFormat fo
             // Gnu BlockDevice: 623 + 1004 + 686 + 1142 = 3455
             return checksum;
         }
+
+        [Fact]
+        void Verify_Size_RegularFile_Empty()
+        {
+            MemoryStream emptyData = new(0);
+            MemoryStream output = new();
+            TarWriter archive = new(output, TarEntryFormat.Pax);
```

Comment:
Did you only see the problem in Pax, or is this the only format that you tested?

Reply:
The one I tested, but the problem would apply equally to all, since small sizes are encoded the same way.